generic/src/main/scala-3/neotypes/generic/deriveCoproductResultMap.scala

```diff
@@ -0,0 +1,22 @@
+package neotypes
+package generic
+
+import shapeless3.deriving.*
+import neotypes.mappers.ResultMapper
+
+trait SealedTraitDerivedCoproductInstances[C] extends ResultMapper.DerivedCoproductInstances[C]
+object SealedTraitDerivedCoproductInstances:
+  given [C](using
+    labelled: Labelling[C],
+    inst: K0.CoproductInstances[ResultMapper, C]
+  ): SealedTraitDerivedCoproductInstances[C] with
+    override def options: List[(String, ResultMapper[C])] =
+      labelled
+        .elemLabels
+        .zipWithIndex
+        .map[(String, ResultMapper[C])] { (s, idx) =>
+          // TODO is there a way to not use asInstanceOf here? widen as it is currently implemented will
+          // not suffice
```

Comment:
Why `widen` does not suffice?

Reply:
If I write `s -> inst.inject(idx) { [t] => (x: ResultMapper[t]) => x.widen[C] }` it raises `Type argument C does not conform to lower bound t`. It seems `C` does not know/could not infer its relationship with `t`.

Reply:
Ah, I found I can inform type bounds here.

https://github.com/typelevel/shapeless-3/blob/4225b010b3fe27b94f194e54cc667ee76f133788/modules/deriving/src/test/scala/shapeless3/deriving/type-classes.scala#L638